pandas/tools/util.py

```diff
@@ -31,14 +31,20 @@ def cartesian_product(X):
     array([1, 2, 1, 2, 1, 2])]
 
     """
```

Comment:
further I think that we *require* a list-like here (maybe its more strict, in that we require an actual array). can you add some tests?